app/service/users.py

```diff
@@ -0,0 +1,42 @@
+import bcrypt
+from fastapi import FastAPI, status
+from fastapi.exceptions import HTTPException
+from sqlmodel import Session, select
+from db import engine
+from models.models import User, UserUpdate
+
+app = FastAPI
+
+def create_user(user:User) -> dict: 
+    with Session(engine) as session:
+        if session.exec(select(User).filter(User.email == user.email)).first():
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail = "user already exist")
+    
+        user.password = bcrypt.hashpw(user.password.encode('utf-8'), bcrypt.gensalt()).decode()
+        session.add(user)
+        session.commit()
+        session.close()
+        return {"message" : "create user"}
+
+def update_user(user_id : int, user:UserUpdate) -> dict:
+    with Session(engine) as session:
+        user = session.exec(select(User).where(User.id == user_id)).one()
+        
+        if not user:
+            raise HTTPException(status_code = status.HTTP_404_NOT_FOUND, detail = "user not exist")
+        
+        session.add(user) 
+        session.commit()
+        session.refresh(user)
+        return {"message" : "user update"}
+
+def remove_user(user_id : int):
+    with Session(engine) as session:
+        user = session.get(User, user_id)
+        if not user:
+            raise HTTPException(status_code = status.HTTP_404_NOT_FOUND, detail = "user not exist")
+        else:
+            session.delete(user)
+            session.commit()
+        return status.HTTP_204_NO_CONTENT
```

Comment:
```suggestion
        session.delete(user)
        session.commit()
        return status.HTTP_204_NO_CONTENT
```
204를 써주신건 너무 잘해주셨습니다. 추가적인 팁은 if 문에서 raise를 하시고 있기 때문에 else를 안쓰셔도 될 것 같습니다.

Reply:
네! tip감사합니다! 수정하였습니다